packages/phone-number-privacy/signer/src/pnp/services/quota.ts

```diff
@@ -5,32 +5,105 @@ import {
   SignMessageRequest,
 } from '@celo/phone-number-privacy-common'
 import { Knex } from 'knex'
-import { incrementQueryCount } from '../../common/database/wrappers/account'
+import { getPerformedQueryCount, incrementQueryCount } from '../../common/database/wrappers/account'
 import { storeRequest } from '../../common/database/wrappers/request'
+import { Counters, Histograms } from '../../common/metrics'
 import { OdisQuotaStatusResult, QuotaService } from '../../common/quota'
+import { getBlockNumber, meter } from '../../common/web3/contracts'
+import { config } from '../../config'
 import { PnpSession } from '../session'
 export interface PnpQuotaStatus {
   queryCount: number
   totalQuota: number
   blockNumber: number
 }
 
-export class PnpQuotaService implements QuotaService<SignMessageRequest | PnpQuotaRequest> {
+// TODO EN: possibly name this OnChainPnpQuotaService, then have LegacyQuotaService & OnChainQuotaService?
+
+export abstract class PnpQuotaService
+  implements QuotaService<SignMessageRequest | PnpQuotaRequest> {
   constructor(readonly db: Knex, readonly kit: ContractKit) {}
 
+  // Differentiating prefix for metering functions
+  protected abstract readonly metricsPrefix: string
+
   public async checkAndUpdateQuotaStatus(
-    _state: PnpQuotaStatus,
-    _session: PnpSession<SignMessageRequest>,
-    _trx: Knex.Transaction
+    state: PnpQuotaStatus,
+    session: PnpSession<SignMessageRequest>,
```

Comment:
This is only expecting a SignMessageRequest and may complain when dealing with a PnpQuotaRequest

Reply:
is there a reason this func would receive a PnpQuotaRequest? I thought that one is strictly for querying quota but not updating state?